src/dual.jl

```diff
@@ -1,68 +1,67 @@
-immutable Dual{T<:Real} <: Number
-    re::T
-    du::T
+typealias ReComp Union{Real,Complex}
+
+immutable Dual{T<:ReComp} <: Number
+    value::T
+    dual::T
```

Comment:
First of all, +1 to this, thanks for writing it up. I'm really interested in seeing this in action.

Can we use `epsilon` instead of `dual` for the `ɛ` component? I feel like it would be clearer, and also less confusing for users transitioning to the new version of DualNumbers.jl (since before, `dual` was the name of the vectorized constructor).